src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -61,4 +69,64 @@ public void hideDetails(Submission submission) {
             }
         }
     }
+
+    /**
+     * Maps abstract Submission type to concrete submission and assigns result.
+     * @param result result that will be assigned to concrete submission
+     * @param concreteSubmission concrete submission that will be mapped from abstract submission
+     * @param <T> concrete submission type
+     * @return concrete submission of type T
+     */
+    public <T extends Submission> T mapAbstractToConcreteSubmission(Result result, T concreteSubmission) {
```

Comment:
As already described in the `FileUploadSubmissionService` I don't think that we need this method. You can just cast the `List<Submission>` to your desired type when fetching all submissions for a result.